master/buildbot/worker/docker.py

```diff
@@ -301,7 +303,8 @@ def _thd_start_instance(self, image, dockerfile, volumes,
             name=self.getContainerName(),
             volumes=volumes,
             environment=self.createEnvironment(),
-            host_config=host_conf
+            host_config=host_conf,
+            hostname=self.hostname
```

Comment:
I think this value should be renderable so that one can define it dynamically according to buildrequest.

you need to add this to renderWorkerProps() and pass the value around